app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,54 @@
+class ReportsController < ApplicationController
+  before_action :authenticate_user!, only: [:create, :new, :edit, :update, :destroy]
+
+  before_action :set_review, only: [:new, :create]
+  before_action :check_admin, only: [:index, :destroy]
+
+  def index
+    @reports = Report.all
+  end
+
+  def new
+    @report = Report.new
+  end
+
+  def create
+    @report = Report.new(report_params)
+    @review.reported = true
```

Comment:
move this part to callback, 
what happens if report does not get saved due to some error, your review would still be marked as reported.